src/virt.c

```diff
@@ -1689,22 +1717,34 @@ static int get_disk_err(virDomainPtr domain) {
 }
 #endif /* HAVE_DISK_ERR */
 
-static int get_block_stats(struct block_device *block_dev) {
+static int get_block_device_stats(struct block_device *block_dev) {
+  virDomainBlockInfo binfo;
+  init_block_info(&binfo);
 
   if (!block_dev) {
     ERROR(PLUGIN_NAME " plugin: get_block_stats NULL pointer");
     return -1;
   }
 
-  struct lv_block_info binfo;
-  init_block_info(&binfo);
+  /* Block info statistics can be only fetched from devices with 'source'
+   * defined */
+  if (block_dev->has_source) {
```

Comment:
Hi,

I think we had to add additional condition check here:
`(extra_stats & (ex_stats_disk_allocation|ex_stats_disk_capacity|ex_stats_disk_physical))`

So, `virDomainGetBlockInfo` will be called only if it's result is really needed and will be used later.

What do you think?

Reply:
yes, that makes sense. Thanks for suggestion:)
I will implement this change with hope that  I will be able to release updated version before code freeze.

Reply:
@rjablonx any chance to include the suggestion?

Reply:
@mrunge, yes of course, it is internally implemented and currently it is under tests (we plan to release it during this week)